proto/buf/alpha/registry/v1alpha1/push.proto

```diff
@@ -39,6 +39,10 @@ message PushRequest {
   repeated string tracks = 6 [deprecated = true];
   // If non-empty, the push creates a draft commit with this name.
   string draft_name = 7;
+  // If set, `module` is ignored and the module's manifest is used instead.
+  buf.alpha.module.v1alpha1.Blob manifest = 8;
+  // blobs to close on the manifest
+  repeated buf.alpha.module.v1alpha1.Blob blobs = 9;
```

Comment:
Correct, but I'm talking about the case of a hash collision, same scenario as [the linked comment](https://github.com/bufbuild/buf/pull/1499#discussion_r998819721), with this request the backend cannot know which blob matches to which file:

```
{
  ...
  blob manifest = """
  shake256:abc123  path/to/file1
  ...
  shake256:abc123  path/to/file2
  """
  []blob blobs = [
    {hash: "shake256:abc123", content: "//contents of file1"},
    ...
    {hash: "shake256:abc123", content: "//contents of file2"},
  ]
}
```

Reply:
I was thinking of creating a manifest with the blob (`manifest.NewManifestFromReader(manifestBlob)`). Then iterate over the blobs by creating a digest for them and calling `manifest.Paths(digest)`.

Reply:
Discussed in zoom: if 2 files have the same digest, it's because either:

- They have the same content (consume contents of whatever match in the array it's fine, it's the same), or
- We've found a collision in the hash, which is highly unlikely, and probably a reason to change hash kind

Note: []blobs will have just one entry per digest, even if many files share that same content.